docs/ci/demo

```diff
@@ -38,6 +38,8 @@ popper run --timeout "1m 10s"
 # arbitrary stages and envs
 init_test
 popper init --stages=one,two,three --envs=host,ubuntu-xenial pipetwo
+popper init --stages=one,two,teardown sample_exp
+
```

Comment:
we can also check for when `teardown` is not the last step and check that an error was thrown.

Reply:
@ivotron Apologies. I was slightly confused on how to handle the stderr output.
I have added it now. Let me know if that is not the correct way to do it.

Reply:
Apologies. I was confused on how to handle the stderr output. 
I have added a test. Please let me know if it requires any changes.